fbcp-ili9341.cpp

```diff
@@ -265,6 +307,15 @@ int main()
       memcpy(framebuffer[0], videoCoreFramebuffer[1], gpuFramebufferSizeBytes);
 #endif
 
+#ifdef LOW_BATT_PIN
+      if (tick() - lowBattLastPolled > LOW_BATT_POLLING_TIME) {
+        lowBatt = GET_GPIO(LOW_BATT_PIN) ? LOW_BATT_STATE : !LOW_BATT_STATE;
+        lowBattLastPolled = tick();
+      }
+      if (lowBatt)
+        drawLowBattIcon(framebuffer[0]);
```

Comment:
There is actually a small gotcha here due to the way the code is structured. The function `DrawLowBatteryIcon(framebuffer[0]);` should be called in two different places, see the locations where the statistics overlay is called with `DrawStatisticsOverlay(framebuffer[0]);`. Battery icon should probably show on top of stats overlay so that the effect of battery icon is observable even when stats overlay is active. (if one is debugging its effects). Perhaps have
```diff
@@ -273,6 +273,7 @@ int main()
       __atomic_fetch_sub(&numNewGpuFrames, numNewFrames, __ATOMIC_SEQ_CST);
 
       DrawStatisticsOverlay(framebuffer[0]);
+      DrawLowBatteryIcon(framebuffer[0]);
 
 #ifdef USE_GPU_VSYNC
 
@@ -292,6 +293,7 @@ int main()
         frameObtainedTime = tick();
         framebufferHasNewChangedPixels = SnapshotFramebuffer(framebuffer[0]);
         DrawStatisticsOverlay(framebuffer[0]);
+        DrawLowBatteryIcon(framebuffer[0]);
         framebufferHasNewChangedPixels = framebufferHasNewChangedPixels && IsNewFramebuffer(framebuffer[0], framebuffer[1]);
       }
 #else
```
and no-op that function from inside if low battery icon is not active. (That way the current ifdef jungle in fbcp-ili9341.cpp file stays a bit lighter)